base/client.jl

```diff
@@ -125,14 +125,26 @@ function repl_cmd(cmd, out)
     nothing
 end
 
-display_error(er) = display_error(er, [])
-function display_error(er, bt)
-    print_with_color(Base.error_color(), STDERR, "ERROR: "; bold = true)
-    with_output_color(Base.error_color(), STDERR) do io
-        showerror(io, er, bt)
+function display_error(io::IO, er, bt)
+    if !isempty(bt)
+        st = stacktrace(bt)
+        if !isempty(st)
+            io = redirect(io, log_error_to, st[1])
+        end
+    end
+    print_with_color(Base.error_color(), io, "ERROR: "; bold = true)
+    # remove REPL-related frames from interactive printing
+    eval_ind = findlast(addr->Base.REPL.ip_matches_func(addr, :eval), bt)
+    if eval_ind != 0
+        bt = bt[1:eval_ind-1]
+    end
+    with_output_color(Base.error_color(), io) do io
```

Comment:
yet this line remains precisely what used to be in `client.jl`;  i've left this line unchanged.

IIRC a few months ago the two copies of `display_error` (in `client.jl` and `REPL.jl`) where identical.  there has been more than one occasion where someone changed one, but not the other, when i think they should've changed both, and either forgot to or were unaware of the other copy.

please let me know if i am wrong in thinking that the two copies shouldn't behave identically.  if this is not the case, then this PR should be modified so as not to delete the one in `REPL.jl`.